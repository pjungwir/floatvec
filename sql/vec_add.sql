-- float add to scalar lhs
SELECT vec_add(5::float, floats) FROM measurements WHERE sensor_id IN (4);

-- float add to scalar rhs
SELECT vec_add(floats, 5::float) FROM measurements WHERE sensor_id IN (4);

-- float add to array
SELECT vec_add(floats, floats) FROM measurements WHERE sensor_id IN (4);

-- float add to null scalar lhs
SELECT vec_add(NULL::float, floats) FROM measurements WHERE sensor_id IN (4);

-- float add to null scalar rhs
SELECT vec_add(floats, NULL::float) FROM measurements WHERE sensor_id IN (4);

-- float add scalar to null array lhs
SELECT vec_add(5::float, NULL::float[]) FROM measurements WHERE sensor_id IN (4);

-- float add scalar to null array rhs
SELECT vec_add(NULL::float[], 5::float) FROM measurements WHERE sensor_id IN (4);

-- float add to null array lhs
SELECT vec_add(floats, NULL::float[]) FROM measurements WHERE sensor_id IN (4);

-- float add to null array rhs
SELECT vec_add(NULL::float[], floats) FROM measurements WHERE sensor_id IN (4);

-- null float add to null scalar lhs
SELECT vec_add(NULL::float, NULL::float[]) FROM measurements WHERE sensor_id IN (4);

-- null float add to null scalar rhs
SELECT vec_add(NULL::float[], NULL::float) FROM measurements WHERE sensor_id IN (4);

-- null float add to null array
SELECT vec_add(NULL::float[], NULL::float[]) FROM measurements WHERE sensor_id IN (4);

-- int add to scalar lhs
SELECT vec_add(5::int, ints) FROM measurements WHERE sensor_id IN (4);

-- int add to scalar rhs
SELECT vec_add(ints, 5::int) FROM measurements WHERE sensor_id IN (4);

-- int add to array
SELECT vec_add(ints, ints) FROM measurements WHERE sensor_id IN (4);

-- int add to null scalar lhs
SELECT vec_add(NULL::int, ints) FROM measurements WHERE sensor_id IN (4);

-- int add to null scalar rhs
SELECT vec_add(ints, NULL::int) FROM measurements WHERE sensor_id IN (4);

-- int add scalar to null array lhs
SELECT vec_add(5::int, NULL::int[]) FROM measurements WHERE sensor_id IN (4);

-- int add scalar to null array rhs
SELECT vec_add(NULL::int[], 5::int) FROM measurements WHERE sensor_id IN (4);

-- int add to null array lhs
SELECT vec_add(ints, NULL::int[]) FROM measurements WHERE sensor_id IN (4);

-- int add to null array rhs
SELECT vec_add(NULL::int[], ints) FROM measurements WHERE sensor_id IN (4);

-- null int add to null scalar lhs
SELECT vec_add(NULL::int, NULL::int[]) FROM measurements WHERE sensor_id IN (4);

-- null int add to null scalar rhs
SELECT vec_add(NULL::int[], NULL::int) FROM measurements WHERE sensor_id IN (4);

-- null int add to null array
SELECT vec_add(NULL::int[], NULL::int[]) FROM measurements WHERE sensor_id IN (4);
