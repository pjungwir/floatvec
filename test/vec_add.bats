load test_helper

@test "float add to scalar lhs" {
  result="$(query "SELECT vec_add(5::float, floats) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{6,NULL,7}" ]
}

@test "float add to scalar rhs" {
  result="$(query "SELECT vec_add(floats, 5::float) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{6,NULL,7}" ]
}

@test "float add to array" {
  result="$(query "SELECT vec_add(floats, floats) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{2,NULL,4}" ]
}


@test "float add to null scalar lhs" {
  result="$(query "SELECT vec_add(NULL::float, floats) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{NULL,NULL,NULL}" ]
}

@test "float add to null scalar rhs" {
  result="$(query "SELECT vec_add(floats, NULL::float) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{NULL,NULL,NULL}" ]
}

@test "float add scalar to null array lhs" {
  result="$(query "SELECT vec_add(5::float, NULL::float[]) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "float add scalar to null array rhs" {
  result="$(query "SELECT vec_add(NULL::float[], 5::float) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "float add to null array lhs" {
  result="$(query "SELECT vec_add(floats, NULL::float[]) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "float add to null array rhs" {
  result="$(query "SELECT vec_add(NULL::float[], floats) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}


@test "null float add to null scalar lhs" {
  result="$(query "SELECT vec_add(NULL::float, NULL::float[]) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "null float add to null scalar rhs" {
  result="$(query "SELECT vec_add(NULL::float[], NULL::float) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "null float add to null array" {
  result="$(query "SELECT vec_add(NULL::float[], NULL::float[]) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}


@test "int add to scalar lhs" {
  result="$(query "SELECT vec_add(5::int, ints) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{6,NULL,7}" ]
}

@test "int add to scalar rhs" {
  result="$(query "SELECT vec_add(ints, 5::int) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{6,NULL,7}" ]
}

@test "int add to array" {
  result="$(query "SELECT vec_add(ints, ints) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{2,NULL,4}" ]
}


@test "int add to null scalar lhs" {
  result="$(query "SELECT vec_add(NULL::int, ints) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{NULL,NULL,NULL}" ]
}

@test "int add to null scalar rhs" {
  result="$(query "SELECT vec_add(ints, NULL::int) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{NULL,NULL,NULL}" ]
}

@test "int add scalar to null array lhs" {
  result="$(query "SELECT vec_add(5::int, NULL::int[]) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "int add scalar to null array rhs" {
  result="$(query "SELECT vec_add(NULL::int[], 5::int) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "int add to null array lhs" {
  result="$(query "SELECT vec_add(ints, NULL::int[]) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "int add to null array rhs" {
  result="$(query "SELECT vec_add(NULL::int[], ints) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}


@test "null int add to null scalar lhs" {
  result="$(query "SELECT vec_add(NULL::int, NULL::int[]) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "null int add to null scalar rhs" {
  result="$(query "SELECT vec_add(NULL::int[], NULL::int) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "null int add to null array" {
  result="$(query "SELECT vec_add(NULL::int[], NULL::int[]) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}


