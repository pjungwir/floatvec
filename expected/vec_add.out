-- float add to scalar lhs
SELECT vec_add(5::float, floats) FROM measurements WHERE sensor_id IN (4);
  vec_add   
------------
 {6,NULL,7}
(1 row)

-- float add to scalar rhs
SELECT vec_add(floats, 5::float) FROM measurements WHERE sensor_id IN (4);
  vec_add   
------------
 {6,NULL,7}
(1 row)

-- float add to array
SELECT vec_add(floats, floats) FROM measurements WHERE sensor_id IN (4);
  vec_add   
------------
 {2,NULL,4}
(1 row)

-- float add to null scalar lhs
SELECT vec_add(NULL::float, floats) FROM measurements WHERE sensor_id IN (4);
     vec_add      
------------------
 {NULL,NULL,NULL}
(1 row)

-- float add to null scalar rhs
SELECT vec_add(floats, NULL::float) FROM measurements WHERE sensor_id IN (4);
     vec_add      
------------------
 {NULL,NULL,NULL}
(1 row)

-- float add scalar to null array lhs
SELECT vec_add(5::float, NULL::float[]) FROM measurements WHERE sensor_id IN (4);
 vec_add 
---------
 
(1 row)

-- float add scalar to null array rhs
SELECT vec_add(NULL::float[], 5::float) FROM measurements WHERE sensor_id IN (4);
 vec_add 
---------
 
(1 row)

-- float add to null array lhs
SELECT vec_add(floats, NULL::float[]) FROM measurements WHERE sensor_id IN (4);
 vec_add 
---------
 
(1 row)

-- float add to null array rhs
SELECT vec_add(NULL::float[], floats) FROM measurements WHERE sensor_id IN (4);
 vec_add 
---------
 
(1 row)

-- null float add to null scalar lhs
SELECT vec_add(NULL::float, NULL::float[]) FROM measurements WHERE sensor_id IN (4);
 vec_add 
---------
 
(1 row)

-- null float add to null scalar rhs
SELECT vec_add(NULL::float[], NULL::float) FROM measurements WHERE sensor_id IN (4);
 vec_add 
---------
 
(1 row)

-- null float add to null array
SELECT vec_add(NULL::float[], NULL::float[]) FROM measurements WHERE sensor_id IN (4);
 vec_add 
---------
 
(1 row)

-- int add to scalar lhs
SELECT vec_add(5::int, ints) FROM measurements WHERE sensor_id IN (4);
  vec_add   
------------
 {6,NULL,7}
(1 row)

-- int add to scalar rhs
SELECT vec_add(ints, 5::int) FROM measurements WHERE sensor_id IN (4);
  vec_add   
------------
 {6,NULL,7}
(1 row)

-- int add to array
SELECT vec_add(ints, ints) FROM measurements WHERE sensor_id IN (4);
  vec_add   
------------
 {2,NULL,4}
(1 row)

-- int add to null scalar lhs
SELECT vec_add(NULL::int, ints) FROM measurements WHERE sensor_id IN (4);
     vec_add      
------------------
 {NULL,NULL,NULL}
(1 row)

-- int add to null scalar rhs
SELECT vec_add(ints, NULL::int) FROM measurements WHERE sensor_id IN (4);
     vec_add      
------------------
 {NULL,NULL,NULL}
(1 row)

-- int add scalar to null array lhs
SELECT vec_add(5::int, NULL::int[]) FROM measurements WHERE sensor_id IN (4);
 vec_add 
---------
 
(1 row)

-- int add scalar to null array rhs
SELECT vec_add(NULL::int[], 5::int) FROM measurements WHERE sensor_id IN (4);
 vec_add 
---------
 
(1 row)

-- int add to null array lhs
SELECT vec_add(ints, NULL::int[]) FROM measurements WHERE sensor_id IN (4);
 vec_add 
---------
 
(1 row)

-- int add to null array rhs
SELECT vec_add(NULL::int[], ints) FROM measurements WHERE sensor_id IN (4);
 vec_add 
---------
 
(1 row)

-- null int add to null scalar lhs
SELECT vec_add(NULL::int, NULL::int[]) FROM measurements WHERE sensor_id IN (4);
 vec_add 
---------
 
(1 row)

-- null int add to null scalar rhs
SELECT vec_add(NULL::int[], NULL::int) FROM measurements WHERE sensor_id IN (4);
 vec_add 
---------
 
(1 row)

-- null int add to null array
SELECT vec_add(NULL::int[], NULL::int[]) FROM measurements WHERE sensor_id IN (4);
 vec_add 
---------
 
(1 row)

